### PR TITLE
Use git wire protocol v2

### PR DESCRIPTION
**Pull request by nicksnyder**
fixes https://github.com/sourcegraph/sourcegraph/issues/699 

Also increases the timeout for ls-remote in case there are customers who don't have a v2 git server yet.

### CODE REVIEW

**Comment by chrismwendt** [comment]
It looks like this build fails every once in a while due to Cypress e2e tests that were removed in master. Rebasing on the latest master will probably make that step pass.

File: cmd/gitserver/server/server.go [hunk 1]
```diff
@@ -975,9 +976,12 @@ func (s *Server) isCloneable(ctx context.Context, url string) error {
 		return testRepoExists(ctx, url)
 	}
 
-	cmd := exec.CommandContext(ctx, "git", "ls-remote", url, "HEAD")
+	cmd := exec.CommandContext(ctx, "git", args...)
 	out, err := s.runWithRemoteOpts(ctx, cmd, nil)
 	if err != nil {
+		if ctxerr := ctx.Err(); ctxerr != nil {
```
Comment (beyang) [inline cmd/gitserver/server/server.go]: Why is this necessary? If the call failed due to a context-related error, wouldn't `err` already be equivalent to `ctx.Err()`?

Comment (nicksnyder) [reply cmd/gitserver/server/server.go]: In general, you can't depend on a function returning the context's error because nothing enforces that.

For this case specifically, it doesn't as evidenced by the attached logs in the linked issue (e.g. `signal: killed`). The running command just reports that it was killed (it doesn't know why), but the real reason is that the context was canceled.
